### ISSUE TITLE
CreateSameFileAtOnceTest.testCreateSameFileAtOnce fails

### ISSUE THREAD

**Issue by m273d15** [saros-project/saros#854]
The test AddMultipleFilesTest.testAddMultipleFilesSimultaneouslyTest fails because of a regression.

**Comment by m273d15** [comment]
The test-case shows the following behavior:
* ALICE starts the session with BOB & CARL (a project with a file "dummy.sync").
* The NetworkManipulator blocks the messages of BOB & CARL
* Both create the file readme.txt
* The NetworkManipulator unblocks the messages
* This results in a session where BOB & CARL have a project with two files and ALICE has only the "dummy.sync" file.
* The test fails, because ALICE waits for the readme file (leads to a timeout exception)

I removed CARL from the test setup and the result is as before (ALICE has one file, Bob two, this leads to a timeout exception).

After removing the NetworkManipulator calls the test is succesful. I noticed that ALICE throws the exception `java.io.IOException: failed to inflate data` (Also included in the logs of the stf runs on the CI server). Therefore, I assume that ALICE is not able to receive the previously blocked message containing the corresponding FileActivity.
<details>
<summary>Exception</summary>
<code>
TRACE 16:20:32,753 [BinaryChannel-jenkins_carl_stf] (BinaryChannelConnection.java:308) processing opcode 0xFA [TFD]: id=0, chunks=1
TRACE 16:20:32,753 [BinaryChannel-jenkins_carl_stf] (BinaryChannelConnection.java:346) processing opcode 0xFB [DATA]: id=0, DATA len=273 bytes
TRACE 16:20:32,753 [saros-DispatchContext] (NetworkManipulatorImpl.java:99) intercepting incoming packet from: jenkins_carl_stf@stf_prosody_test.org/Saros
TRACE 16:20:32,753 [saros-DispatchContext] (XMPPReceiver.java:214) received binary XMPP extension: TransferDescription [elementName=ados, namespace=saros, recipient=jenkins_alice_stf@stf_prosody_test.org/Saros, sender=jenkins_carl_stf@stf_prosody_test.org/Saros, compress=true], size: 273, RX time: 0 ms [SOCKS5 (D)]
ERROR 16:20:32,753 [saros-DispatchContext] (XMPPReceiver.java:232) could not decompress extension payload
java.io.IOException: failed to inflate data
	at saros.net.internal.XMPPReceiver.inflate(XMPPReceiver.java:324)
	at saros.net.internal.XMPPReceiver.convertBinaryXMPPExtension(XMPPReceiver.java:230)
	at saros.net.internal.XMPPReceiver.access$200(XMPPReceiver.java:37)
	at saros.net.internal.XMPPReceiver$6.run(XMPPReceiver.java:169)
	at saros.util.ThreadUtils$1.run(ThreadUtils.java:34)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.zip.DataFormatException: incorrect header check
	at java.util.zip.Inflater.inflateBytes(Native Method)
	at java.util.zip.Inflater.inflate(Inflater.java:259)
	at java.util.zip.Inflater.inflate(Inflater.java:280)
	at saros.net.internal.XMPPReceiver.inflate(XMPPReceiver.java:319)
	... 9 more
</code>
</details>

I performed a manual bisect and found that:
* Test always fails with the same exception with commit 23529f7fb31f523b4ab808984ba37226ad5a82bc "[INTERNAL] Moves send logic from DTM to XMPPTransmitter". (This commit still requires an eclipse neon for stf testing)
* Test is succesful with commit b24e8ea4d267a6cff317c720e4fa92fc4060b9b1 "[UI][I] Use IntelliJ UI wrapper classes" (This commit still requires an eclipse neon for stf testing)

It seems like PR #652 introduced this issue @srossbach do you know whether this could have led to an incompatibility of the NetworkManipulator and the productive extension handling?

**Comment by srossbach** [comment]
I will take a look at it.

**Comment by srossbach** [comment]
Ok I found the issue. The compression flag is set but the compression is not done.

**Comment by srossbach** [comment]
I am currently unable to start the STF.

Every test case stops instant with:

```
java.lang.NullPointerException
	at org.eclipse.ui.internal.ViewReference.getPartName(ViewReference.java:69)
	at org.eclipse.swtbot.eclipse.finder.widgets.SWTBotWorkbenchPart.getTitle(SWTBotWorkbenchPart.java:133)
	at saros.stf.server.rmi.remotebot.impl.RemoteWorkbenchBot.getTitlesOfOpenedViews(RemoteWorkbenchBot.java:98)
	at saros.stf.server.rmi.remotebot.impl.RemoteWorkbenchBot.isViewOpen(RemoteWorkbenchBot.java:104)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at sun.rmi.server.UnicastServerRef.dispatch(Unknown Source)
	at sun.rmi.transport.Transport$1.run(Unknown Source)
	at sun.rmi.transport.Transport$1.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.Transport.serviceCall(Unknown Source)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(Unknown Source)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(Unknown Source)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
	at sun.rmi.transport.StreamRemoteCall.exceptionReceivedFromServer(Unknown Source)
	at sun.rmi.transport.StreamRemoteCall.executeCall(Unknown Source)
	at sun.rmi.server.UnicastRef.invoke(Unknown Source)
	at java.rmi.server.RemoteObjectInvocationHandler.invokeRemoteMethod(Unknown Source)
	at java.rmi.server.RemoteObjectInvocationHandler.invoke(Unknown Source)
	at com.sun.proxy.$Proxy9.isViewOpen(Unknown Source)
	at saros.stf.client.StfTestCase.setUpWorkbench(StfTestCase.java:289)
	at saros.stf.client.StfTestCase.select(StfTestCase.java:169)
	at saros.stf.test.stf.basicwidget.BasicWidgetTreeTest.selectTesters(BasicWidgetTreeTest.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)

```

Did I missed some changes (have not run gradle for a while). I am currently using Eclipse Neon and the used SWTBot version should work for this Eclipse release.

**Comment by tobous** [comment]
@srossbach I had the same issue. For me, the issue went away when deleting the local test workspaces (`workspace-alice-stf`, `workspace-bob-stf`, etc.) and working dirs (`working-dir-alice-stf`, etc.).

And I thing you need Eclipse Photon (4.8). At least that version is confirmed working. I don't know whether Neon works as well. The update to 4.8 was made due to some dependency issues with the Eclipse Maven artifacts a while ago. See issue #795 and fix PR #800.

**Comment by tobous** [comment]
As the test was fixed, this can be closed and the test annotation can be removed.
